src/test/java/com/palantir/gradle/gitversion/JGitDescribeTest.java

```diff
@@ -0,0 +1,223 @@
+package com.palantir.gradle.gitversion;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.MergeCommand;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.lib.PersonIdent;
+import org.eclipse.jgit.lib.Ref;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class JGitDescribeTest {
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    private File projectDir;
+    private Git git;
+    private PersonIdent identity = new PersonIdent("name", "email@address");
+
+    @Before
+    public void before() throws GitAPIException {
+        projectDir = temporaryFolder.getRoot();
+        git = Git.init().setDirectory(projectDir).call();
+    }
+
+    @Test
+    public void test_on_annotated_tag() throws Exception {
```

Comment:
Should we also be deleting the corresponding tests from the gradle integration tests?

Reply:
Or do we want to leave the test coverage there?  (e.g. symlinks and dirty tree?)

Reply:
I kind of see them more as integration tests. They actually apply plugin and execute gradle tasks so I'd keep them.